MetaMask/metamask-extension
# Update Snaps locale messages for casing and content

## Explanation

Updates Snaps i18n messages for casing and (some) content. Snaps casing rules:
- We refer to the platform as "MetaMask Snaps" or just "Snaps", i.e. always capitalized
  - **In general, whenever we refer to the platform, we should use the full name "MetaMask Snaps" at least once before using the shorthand "Snaps".**
- Individual snaps are "snaps" plural and "snap" singular, i.e. never capitalized

In addition, Snaps-related prompts (installation, updating, etc.) were modified to match the title capitalization of other confirmations (i.e., they are not formally title capitalized, and we only capitalize the first word. See e.g. "Permission request" and "Signature request").

## Manual Testing Steps

Check our English locale messages to see if I caught all mis-capitalized cases. Non-English locale messages were only updated via linting (some descriptions changed).

## Pre-merge reviewer checklist
